src/toil/provisioners/aws/awsProvisioner.py

```diff
@@ -1376,29 +1659,82 @@ def kubernetes_policy(self) -> dict:
             "kms:DescribeKey"
         ])])
 
+    def _setup_iam_ec2_role(self, local_role_name: str, policies: Dict[str, Any]) -> str:
+        """
+        Create an IAM role with the given policies, using the given name in
+        addition to the cluster name, and return its full name.
+        """
+
+        # Grab the connection we need to use for this operation.
+        iam = self.aws.boto2(self._region, 'iam')
+
+        # Make sure we can tell our roles apart from roles for other clusters
+        aws_role_name = self._namespace_name(local_role_name)
+        try:
+            # Make the role
+            logger.debug('Creating IAM role...')
+            iam.create_role(aws_role_name, assume_role_policy_document=json.dumps({
+                "Version": "2012-10-17",
+                "Statement": [{
+                    "Effect": "Allow",
+                    "Principal": {"Service": ["ec2.amazonaws.com"]},
+                    "Action": ["sts:AssumeRole"]}
+                ]}))
+            logger.debug('Created new IAM role')
+        except BotoServerError as e:
+            if e.status == 409 and e.error_code == 'EntityAlreadyExists':
+                logger.debug('IAM role already exists. Reusing.')
+                pass
+            else:
+                raise
+
+        # Delete superfluous policies
+        policy_names = set(iam.list_role_policies(aws_role_name).policy_names)
```

Comment:
Hmmm... I remember Asha added a way to supply policies from the user side.  Is it possible for this to unexpectedly prune a policy that, if supplied manually, might break on something else?  I believe it was "--awsEc2ProfileArn": https://github.com/DataBiosphere/toil/pull/2703/files